config/config.go

```diff
@@ -12,27 +12,36 @@ import (
 
 // S3CloudStorageConfig stores the configuration of an S3 API proxy backend.
 type S3CloudStorageConfig struct {
-	Endpoint        string `yaml:"endpoint"`
-	Bucket          string `yaml:"bucket"`
-	Prefix          string `yaml:"prefix"`
-	AccessKeyID     string `yaml:"access_key_id"`
-	SecretAccessKey string `yaml:"secret_access_key"`
-	DisableSSL      bool   `yaml:"disable_ssl"`
-	IAMRoleEndpoint string `yaml:"iam_role_endpoint"`
-	Region          string `yaml:"region"`
-	KeyVersion      int    `yaml:"key_version"`
+	Endpoint        string                `yaml:"endpoint"`
+	Bucket          string                `yaml:"bucket"`
+	Prefix          string                `yaml:"prefix"`
+	AccessKeyID     string                `yaml:"access_key_id"`
+	SecretAccessKey string                `yaml:"secret_access_key"`
+	DisableSSL      bool                  `yaml:"disable_ssl"`
+	IAMRoleEndpoint string                `yaml:"iam_role_endpoint"`
+	Region          string                `yaml:"region"`
+	KeyVersion      int                   `yaml:"key_version"`
+	Uploaders       *ProxyUploadersConfig `yaml:"uploaders"`
 }
 
 // GoogleCloudStorageConfig stores the configuration of a GCS proxy backend.
 type GoogleCloudStorageConfig struct {
-	Bucket                string `yaml:"bucket"`
-	UseDefaultCredentials bool   `yaml:"use_default_credentials"`
-	JSONCredentialsFile   string `yaml:"json_credentials_file"`
+	Bucket                string                `yaml:"bucket"`
+	UseDefaultCredentials bool                  `yaml:"use_default_credentials"`
+	JSONCredentialsFile   string                `yaml:"json_credentials_file"`
+	Uploaders             *ProxyUploadersConfig `yaml:"uploaders"`
 }
 
 // HTTPBackendConfig stores the configuration for a HTTP proxy backend.
 type HTTPBackendConfig struct {
-	BaseURL string `yaml:"url"`
+	BaseURL   string                `yaml:"url"`
+	Uploaders *ProxyUploadersConfig `yaml:"uploaders"`
+}
+
+// ProxyUploadersConfig stores the configuration for uploaders and queue size in a Proxy backend.
+type ProxyUploadersConfig struct {
+	Num       int `yaml:"num_uploaders"`
```

Comment:
I wonder if we should just make these top-level configuration settings since they're identical for each proxy backend? Something like `num_proxy_uploaders` and `max_proxy_queue_size` might be reasonable names.

Reply:
I thought about that, but the benefit of using a struct is we can check if it's nil and provide the default values (unless the yaml package provides something for that) that have been hard coded. Unfortunately the zero values work against us here and I didn't want users to unwittingly turn off uploads. 

Happy to change this though if there's another solution!

Reply:
In `newFromYaml` we pass an empty `Config` to `yaml.Unmarshal`, and values are only set in the `Config` if they exist in the yaml file. Instead of passing an empty `Config` I think we can pass in one with defaults set.